src/main/kotlin/config/GithubConfig.kt

```diff
@@ -0,0 +1,112 @@
+package org.openstreetmap.josm.gradle.plugin.config
+
+import org.gradle.api.Project
+import org.openstreetmap.josm.gradle.plugin.ghreleases.DEFAULT_GITHUB_API_URL
+import org.openstreetmap.josm.gradle.plugin.ghreleases.DEFAULT_GITHUB_UPLOAD_URL
+import org.openstreetmap.josm.gradle.plugin.ghreleases.DEFAULT_GITHUB_URL
+import org.openstreetmap.josm.gradle.plugin.task.GithubReleaseTaskException
+import java.io.File
+import java.io.IOException
+
+/**
+ * Configuration options for the new GitHub releases feature.
+ *
+ * Note: This is currently in beta stage, so expect sudden changes to this API anytime.
+ */
+class GithubConfig(project: Project) {
```

Comment:
@Gubaer I added this class to allow configuration like this in the `build.gradle`:
```gradle
josm {
  github {
    accessToken = "abcdefg"
    …
  }
}
```
I'd like to avoid configuration through Gradle properties altogether. For the manifest this is still possible for backwards compatibility with the Ant build, but environment variables and this new way of configuration should be enough.